holoviews/plotting/bokeh/renderer.py

```diff
@@ -208,10 +215,17 @@ def server_doc(self_or_cls, obj, doc=None):
 
     def _figure_data(self, plot, fmt='html', doc=None, **kwargs):
         model = plot.state
-        doc = Document() if doc is None else doc
+        if plot.document:
+            doc = plot.document
+        else:
+            doc = Document() if doc is None else doc
```

Comment:
This is internal API now and I don't think ``doc`` should be passed into this method at all now. We were already discussing providing a public API that does something similar. I'll have to look at some paramnb/parambokeh apps that use this codepath to figure out the best approach for migrating to the new API without breaking old apps.

Reply:
That makes sense though I think that should be in a PR about refactoring, not one adding a new feature...